libdnf/transaction/Swdb.cpp

```diff
@@ -58,6 +58,11 @@ Swdb::Swdb(const std::string &path)
         // writing to an in-memory database
         conn = std::make_shared< SQLite3 >(path);
         Transformer::createDatabase(conn);
+    } else if (!pathExists(path.c_str()) && geteuid() != 0) {
```

Comment:
What about to use a different approach like to test write permission in given path?

Also please could you avoid double call of `pathExists(path.c_str())`?

Reply:
I have reordered the conditions to call pathExists() just once.

Reply:
> What about to use a different approach like to test write permission in given path?
Let's stick with root user for now. We'd have to check unix perms, access control lists and much more and it still wouldn't be perfect.